### PR TITLE
Feature/tr 1252 handle stalled event

### PR DESCRIPTION
**Pull request by btamas**
Related to https://oat-sa.atlassian.net/browse/TR-1252

- Listen to video stalled state
- Show button to make video reloadable by user
- Tests for stalled behavior

Test:
- Have an item with media interaction and have another one with media player shared stimulus
- Play video from CDN and disconnect
- You should see a message and reload button that reloads the video when you are online again
- You should not see the message when your internet is on, but it is slow

### CODE REVIEW

**Comment by marpesia** [comment]
If I block the URL of the media, it returns a general error and goes back to the list of tests. That's correct?
image

**Comment by btamas** [comment]
> If I block the URL of the media, it returns a general error and goes back to the list of tests. That's correct?

I think yes, because the item is not loaded correctly, so it is on the item level not on the interaction level.

**Comment by marpesia** [comment]
Approved

**Review by marpesia — APPROVED** [review]
Great job! Very detail :)

**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Pull request's target is not `master`
- [ ] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ]  Commits messages are meaningful
- [ ]  Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by jsconan — APPROVED** [review]
The code looks good, nice work.

Unfortunately, I was not able to fully test the feature.

I have 2 small comments, but not blocking

**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by meefox — APPROVED** [review]
Looking good, please check small notes from JS.
Screenshot 2021-06-02 at 16 27 06

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: test/mediaplayer/test.js [hunk 1]
```diff
@@ -1039,4 +1039,46 @@ define(['jquery', 'lodash', 'ui/mediaplayer'], function($, _, mediaplayer) {
             player.destroy();
         });
 
+    QUnit.only('stalled', function(assert) {
```
[the remark this answers is not in the document]
Comment (btamas) [reply test/mediaplayer/test.js:1042]: I used it for debug purpose. Removed.

File: src/mediaplayer/scss/player.scss [hunk 2]
```diff
@@ -98,6 +96,30 @@ $controlsHeight: 36px;
                     opacity: 0.6;
                 }
             }
+
+            &.reload {
+                width: 100%;
+                font-size: 50px;
+                line-height: 30px;
```
[the remark this answers is not in the document]
Comment (btamas) [reply src/mediaplayer/scss/player.scss:103]: I checked how the icon was implemented, and it was pixels, so I used pixels to overwrite them.

**Pull request merged**